### PR TITLE
Add Hapi

### PR DESCRIPTION
**Pull request by MAfarrag**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by MAfarrag** [comment]
@conda-forge/staged-recipes please review

**Comment by MAfarrag** [comment]
Ready for review @conda-forge/help-python

File: recipes/Hapi/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    - shapely
+    - statsmodels
+    - rasterio
+    - rasterstats
+    - netCDF4
+    - scikit-learn
+    - scikit-image
+    - ecmwf-api-client
+    - oasis
+    - joblib
```
Comment (ocefpaf) [inline recipes/Hapi/meta.yaml:42]: You probably only need python and pip here. Unless the package imports itself in the setup.py. If that is the case please open an issue upstream b/c that usually leads to bad things happening.

File: recipes/Hapi/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+build:
+  number: 0
+  noarch: python
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+
+  host:
+    - python >=2.7
```
Comment (ocefpaf) [inline recipes/Hapi/meta.yaml:20]: Conda-forge only builds >=3.6. No more legacy Python ;-p

File: recipes/Oasis/meta.yaml [hunk 3]
```diff
@@ -1,44 +0,0 @@
-{% set name = "Oasis" %}
```
Comment (ocefpaf) [inline recipes/Oasis/meta.yaml:1]: We recommend to not delete recipes b/c that may prevent a feedstock from getting created. Is this from another PR? If so please restore it.

Comment (MAfarrag) [reply recipes/Oasis/meta.yaml]: ok I restored it, now i have to keep both libraries in the recipes all the time ?

File: recipes/Hapi/meta.yaml (hunk 2 above)
Comment (MAfarrag) [reply recipes/Hapi/meta.yaml]: ok updated to 3.6

Comment (MAfarrag) [reply recipes/Hapi/meta.yaml]: However, in my previous recipe (oasis) in conda-forge, I considered python 2.7 and it went through

File: recipes/Hapi/meta.yaml (hunk 1 above)
Comment (MAfarrag) [reply recipes/Hapi/meta.yaml:42]: @conda-forge/help-python  @conda-forge/help-python-c so what should I do here ?, if I remove those packages the recipe will not pass the test they are needed in the package

Comment (ocefpaf) [reply recipes/Hapi/meta.yaml:42]: `run` are the package requirementes, host are build requirements, I'm sure those are not needed to build it, unless the package imports itself in setup.py. Let's test it...

Comment (MAfarrag) [reply recipes/Hapi/meta.yaml]: @ocefpaf it passed the checks

**Comment by MAfarrag** [comment]
it passed the checks @ocefpaf can you review it 
- What about old recipes, I have another previous package, do I have to keep it in the recipe the whole time?
- in this previous package (it was added to conda-forge a couple of days ago) I have python>2.7 and it passed already, did it actually considered checking p2.7 or just considered 3.6 and above
thanks

**Comment by ocefpaf** [comment]
> * What about old recipes, I have another previous package, do I have to keep it in the recipe the whole time?

Sorry but that question is not clear to me. Do you have another feedstock with another package? Keep it in which recipe?

> * in this previous package (it was added to conda-forge a couple of days ago) I have python>2.7 and it passed already, did it actually considered checking p2.7 or just considered 3.6 and above

Ah. The py27 was never built b/c we no longer do that but if it is noarch it can be isntalled in py27. However, it is not guaranteed to work, so we ask people to use >=3.6 from now on.

**Comment by MAfarrag** [comment]
@ocefpaf sorry the feedstock is empty here [https://github.com/conda-forge/hapi-feedstock](https://github.com/conda-forge/hapi-feedstock)

**Pull request merged**